### PR TITLE
Proposal: Create a new "Per LabelSet" limit

### PR DESCRIPTION
**Pull request by alanprot**
**What this PR does**:

This pull request (PR) introduces a new limit Active Series "Per LabelSet." This limit functions similarly to -ingester.max-series-per-metric, but allowing users to define the maximum number of series per LabelSet.

These limits are configurable via runtime settings and with the following format:

```
max_series_per_label_set:
  - label_set:
      labelName1: LabelValue1
    limit: 10
  - label_set:
      labelName1: LabelValue1
      labelName2: LabelValue2
    limit: 3
```

The limits are applied "Per labelset", meaning that only metrics containing all the labels of the limit will be subjected to it.

Additionally, a new metric is implemented to monitor the active timeseries for each limit as follows:

```
# HELP cortex_ingester_active_series_per_labelset Number of currently active series per user and labelset.
# TYPE cortex_ingester_active_series_per_labelset gauge
cortex_ingester_active_series_per_labelset{labelset="{label1=\"value1\"}",user="1"} 3
cortex_ingester_active_series_per_labelset{labelset="{label2=\"value2\"}",user="1"} 2
cortex_ingester_active_series_per_labelset{labelset="{comp1=\"compValue1\", comp2=\"compValue2\"}",user="1"} 2
cortex_ingester_active_series_per_labelset{labelset="{comp1=\"compValue1\"}",user="1"} 7
cortex_ingester_active_series_per_labelset{labelset="{comp2=\"compValue2\"}",user="1"} 2
```

The motivation behind these limits is to enable service administrators to set constraints on a per labelset basis. This functionality helps mitigate the risk of unexpected increases in the number of timeseries.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [X] Tests updated
- [X] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by alanprot** [comment]
I still need to update the changelog but I think we can start reviewing it.

cc @yeya24 @friedrich-at-adobe

**Review by yeya24 — APPROVED** [review]
Thanks

File: pkg/ingester/ingester.go [hunk 1]
```diff
@@ -1100,38 +1117,43 @@ func (i *Ingester) Push(ctx context.Context, req *cortexpb.WriteRequest) (*corte
...
 
-			case errMaxSeriesPerMetricLimitExceeded:
+			case errors.Is(cause, errMaxSeriesPerMetricLimitExceeded):
 				perMetricSeriesLimitCount++
 				updateFirstPartial(func() error {
 					return makeMetricLimitError(perMetricSeriesLimit, copiedLabels, i.limiter.FormatError(userID, cause))
 				})
 				continue
+			case errors.As(cause, &errMaxSeriesPerLabelSetLimitExceeded{}):
+				updateFirstPartial(func() error {
```
Comment (danielblando) [inline pkg/ingester/ingester.go:1153]: Dont we want to create a new discarded samples reason label here?

Comment (alanprot) [reply pkg/ingester/ingester.go:1153]: indeed.

I will create another PR with this and the chngelog in a bit.

**Comment by jeromeinsf** [comment]
Should https://cortexmetrics.io/docs/configuration/configuration-file/ be updated?

**Pull request merged**